sklearn/utils/estimator_checks.py

```diff
@@ -2448,4 +2448,12 @@ def check_fit_idempotent(name, estimator_orig):
     for method in check_methods:
         if hasattr(estimator, method):
             new_result = getattr(estimator, method)(X_test)
-            assert_allclose_dense_sparse(result[method], new_result)
+            if np.issubdtype(new_result.dtype, np.floating):
+                tol = np.finfo(new_result.dtype).eps
```

Comment:
No actually for float64, this would `2.22e-16` do we actually need such precision? I think the previous default of 1e-7 was enough? For other dtypes, I agree.

Maybe then,
```
rtol = max(tol, 1e-7)
atol = tol
```

also maybe take `2*np.finfo(dtype).eps` just to be safe?

Reply:
Thanks for the good feedback. I agree, and have pushed both proposed changes.